### PR TITLE
Validate VarianceScaling distribution type

### PR DESCRIPTION
**Pull request by CHENNA-NAGESH**
This PR improves input validation for `tf.keras.initializers.VarianceScaling` by adding a clear error message when `distribution` is not a string.

The change helps users understand the expected input type and includes a focused unit test to verify the behavior.

### CODE REVIEW

**Comment by CHENNA-NAGESH** [comment]
I have signed the Google CLA. Could you please re-run the CLA check?

**Comment by CHENNA-NAGESH** [comment]
I checked the CUDA13 failure logs. The root cause appears to be a CUDA CI environment issue:

`CUDA_ERROR_SYSTEM_DRIVER_MISMATCH: system has unsupported display driver / cuda driver combination`

Several GPU tests fail because no GPU device is visible, for example `VisibleDeviceCount() > 0 (-1 vs. 0)` and `devices.size() == 0`.

This PR only changes Keras initializer validation under `tensorflow/python/keras/initializers`, and the passing CPU/Windows/ARM jobs suggest the change itself is not causing these CUDA failures. Could a maintainer please rerun the CUDA jobs or confirm this is unrelated?

**Comment by CHENNA-NAGESH** [comment]
"Hi @keerthanakadiri, thank you for updating the labels! Just letting you know that all the CI builds and tests have successfully passed. Please let me know if you would like me to make any changes or if you have any feedback. Thanks!"

**Comment by CHENNA-NAGESH** [comment]
"Hi @keerthanakadiri, thank you for updating the labels! Just letting you know that all the CI builds and tests have successfully passed. Please let me know if you would like me to make any changes or if you have any feedback. Thanks!"

**Comment by CHENNA-NAGESH** [comment]
"Hi @keerthanakadiri , just checking in to see if you've had a chance to look at this or if there is anything I can do to help move this forward. Thanks!"

**Comment by CHENNA-NAGESH** [comment]
"Hi @cantonios, just checking in to see if you have any feedback or if there's anything I can do to help with the review. All the CI checks and builds are green. Thanks!"

**Comment by CHENNA-NAGESH** [comment]
"Hi @keerthanakadiri, I have updated the code to address the feedback. I've modified the error message to print the argument's type and renamed the test method to follow PEP 8 snake_case. Thanks!"

**Review by dmiltr3 — CHANGES_REQUESTED** [review]
Hi @CHENNA-NAGESH,
  
  Thank you for the contribution! Improving validation for the `distribution` parameter in `VarianceScaling` prevents confusing Python interpreter failures (e.g. `AttributeError` from `.lower()`).
  
  To get this merged, please verify:
  1. **Google CLA:** Ensure you have signed the Google Contributor License Agreement (referenced in the checks).
  2. **Unit Tests:** That the PR includes a focused test case validating that non-string distributions (such as `None` or an integer) properly raise a `ValueError` (or `TypeError`) with your custom message.
  
  If you need a reference for the unit test, you can implement one similar to this within the initializer test suite:
  
  ```python
  def test_variance_scaling_invalid_distribution_type(self):
      # Verify that a custom error is raised when distribution is not a string
      with self.assertRaisesRegex(ValueError, "must be a string"):
          initializers.VarianceScaling(distribution=None)
      with self.assertRaisesRegex(ValueError, "must be a string"):
          initializers.VarianceScaling(distribution=123)
 ```
Once the test is added and the presubmits are green, we will import this internally and merge it.

**Comment by CHENNA-NAGESH** [comment]
"Hi @dmiltr3, thank you for the feedback! I have updated the unit test to verify both None and 123 distribution types as suggested, and successfully pushed the changes. Thanks!"

**Review by dmiltr3 — DISMISSED** [review]

**Review by dmiltr3 — APPROVED** [review]
Thank you for this contribution!

This change looks good and safely prevents `AttributeError` when an invalid type (like `None` or an integer) is passed to the `distribution` argument in `VarianceScaling` initializers.

The added runtime type checks and unit tests correctly address the issue and follow the pattern for argument verification by raising a descriptive `ValueError`.
